internal/proxy/proxy.go

```diff
@@ -109,3 +111,46 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	p.reverseProxy.ServeHTTP(w, r)
 }
+
+type Middleware func(r *http.Request)
+
+func Middlewares(middlewares ...Middleware) func(r *http.Request) {
+	return func(r *http.Request) {
+		for _, m := range middlewares {
+			m(r)
+		}
+	}
+}
+
+func MiddlewareSetUpstream(upstream *url.URL) Middleware {
+	return func(r *http.Request) {
+
+		r.URL.Scheme = upstream.Scheme
+		r.URL.Host = upstream.Host
```

Comment:
We still need to do something like:
`
```suggestion
		r.URL.Scheme = upstream.Scheme
		r.URL.Host = upstream.Host
		r.URL.Path = path.Join(upstream.Path, r.URL.Path)
```